src/app/services/cart.service.ts

```diff
@@ -0,0 +1,37 @@
+import { Injectable } from '@angular/core';
+import { IBook } from '../models/book.model';
+import { ICart } from '../models/cart.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CartService {
+  booksInCart: ICart[] = [];
+
+  getItems(): ICart[] {
+    return this.booksInCart;
+  }
+
+  addToCart(book: IBook): void {
+    const ind: number = this.booksInCart.findIndex((el) => el.book.id === book.id);
```

Comment:
Maybe list is not the best data structure for books? Searching for book would be easier and faster if booksInCart was Map with book id as key